network_simulator/flow.py

```diff
@@ -89,5 +92,218 @@ def handle_packet_success(self, packet_id):
     def __repr__(self):
         return str(self.__dict__)
 
-# TODO(team): Flow-related Event subclasses
-# TODO(team): Flow subclasses for at least two TCP algorithms.
+
+
+class InitiateFlowEvent(Event):
+    def __init__(self, flow=None, packet_ids=[]):
+        """
+        An one-time Event called to set up the Flow at a given timestamp.
+
+        :ivar Flow flow: flow of this event.
+        :ivar list packet_ids: a list of integers of packet IDs.
+        :ivar list packets_to_send: list of data packets to send. This is
+        default to an empty list. This list will be populated later.
+        """
+        self.flow = flow
+        self.packet_ids = packet_ids
+        self.packets_to_send = []
+
+    def create_data_packet(self, packet_id, curr_time):
+        """
+        Create a DataPacket based on the current flow.
+
+        :param int packet_id: corresponding packet ID.
+        :param float curr_time: simulation time of this function call.
+        :return DataPacket
+        """
+        packet = DataPacket(packet_id, self.flow.flow_id,
```

Comment:
Wouldn't a constructor call require specifying the param names since these are "optional" in the constructor? E.g. packet_id=packet_id. I forget how Python does this.